src/platform/nxp/common/ConnectivityManagerImpl.cpp

```diff
@@ -591,10 +596,10 @@ void ConnectivityManagerImpl::ConnectNetworkTimerHandler(::chip::System::Layer *
     }
     else
     {
-        PlatformMgr().LockChipStack();
-        DeviceLayer::SystemLayer().StartTimer(System::Clock::Milliseconds32(CHIP_DEVICE_CONFIG_WIFI_STATION_RECONNECT_INTERVAL),
-                                              ConnectNetworkTimerHandler, context);
-        PlatformMgr().UnlockChipStack();
+        /* Post an event to start a delay timer asynchronously in the Matter task context */
+        event.Type                       = DeviceEventType::kPlatformNxpStartWlanInitWaitTimerEvent;
+        event.Platform.pNetworkDataEvent = (struct wlan_network *) context;
+        (void) PlatformMgr().PostEvent(&event);
```

Comment:
This may need a VerifyOrDie otherwise you may fail to connect on queue full once.

Reply:
Is this suitable ?
(void) PlatformMgr().**_PostEventOrDie_**(&event);